### PR TITLE
fix: extend xts suite with new coverage for HBAR transfers to zero address and reserved system accounts

### LINKED ISSUE #4332: Add Hedera-specific negative tests for eth_sendRawTransaction
### Description

Add negative test coverage for HBAR transfer edge cases specific to Hedera behavior via eth_sendRawTransaction:

* Sending HBARs to the zero address → expect `INVALID_SOLIDITY_ADDRESS`
* Sending HBARs to a long zero address below 359 (e.g., id = 100) → expect `INVALID_FEE_SUBMITTED`
* Sending HBARs to a non-existing long zero address between 750–1000 (e.g., id = 850) → expect `INVALID_ALIAS_KEY`

### Acceptance Criteria

* New tests are added to the acceptance suite and XTS test suite
* Tests verify expected Hedera-specific reverts

### PR DESCRIPTION
**Pull request by quiet-node**
### Description

This PR adds negative test coverage for HBAR transfer edge cases specific to Hedera behavior via eth_sendRawTransaction. The tests verify expected Hedera-specific reverts for:

- Sending HBARs to the zero address → expect INVALID_SOLIDITY_ADDRESS
- Sending HBARs to a long zero address below 359 (e.g., id = 100) → expect INVALID_CONTRACT_ID
- Sending HBARs to a non-existing long zero address between 750–1000 (e.g., id = 850) → expect INVALID_ALIAS_KEY
- Updated contract deployment fee assertion test for [HIP-1249](https://hips.hedera.com/hip/hip-1249) compatibility (eliminates 80% minimum charge rule). This feature was recently merged to HEAD of consensus main in this PR https://github.com/hiero-ledger/hiero-consensus-node/pull/20703.

New tests are added to the acceptance suite and XTS test suite to ensure proper validation of these edge cases.

### Related issue(s)

Fixes #4332

### Testing Guide

1. Run the acceptance test suite to verify the new negative tests: `npm run acceptancetest:xts`
2. Check that the tests correctly expect the specified Hedera-specific revert reasons (INVALID_SOLIDITY_ADDRESS, INVALID_CONTRACT_ID, INVALID_ALIAS_KEY) for the respective edge cases.

### Changes from original design (optional)

N/A

### Additional work needed (optional)

N/A

### Checklist

- [x] I've assigned an assignee to this PR and related issue(s) (if applicable)
- [x] I've assigned a label to this PR and related issue(s) (if applicable)
- [x] I've assigned a milestone to this PR and related issue(s) (if applicable)
- [x] I've updated documentation (code comments, README, etc. if applicable)
- [x] I've done sufficient testing (unit, integration, etc.)

### CODE REVIEW

**Review by natanasow — DISMISSED** [review]

File: packages/server/tests/acceptance/rpc_batch1.spec.ts [hunk 1]
```diff
@@ -987,6 +987,116 @@ describe('@api-batch-1 RPC Server Acceptance Tests', function () {
...
+          '0x0000000000000000000000000000000000000169', // 0.0.361 PRNG
+          '0x000000000000000000000000000000000000016a', // 0.0.362 HAS
+
+          // non-existent accounts
+          '0x00000000000000000000000000000000000001C2', // 0.0.450
+          '0x00000000000000000000000000000000000001FE', // 0.0.510
+          '0x00000000000000000000000000000000000002EE', // 0.0.750
+        ];
+
+        for (const address of hederaSystemAccounts) {
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply packages/server/tests/acceptance/rpc_batch1.spec.ts]: Yes I agree! Fixed

File: packages/server/tests/acceptance/rpc_batch1.spec.ts [hunk 2]
```diff
@@ -987,6 +987,116 @@ describe('@api-batch-1 RPC Server Acceptance Tests', function () {
...
+
+        await Assertions.assertPredefinedRpcError(error, sendRawTransaction, true, relay, [
+          signedSendHbarTx,
+          requestDetails,
+        ]);
+      });
+
+      it('@xts should reject eth_sendRawTransaction requests for HBAR crypto transfers to reserved system account addresses (accounts ≤ 0.0.750) with INVALID_CONTRACT_ID.', async function () {
+        // https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hedera-node/docs/system-accounts-operations.md
+        const hederaSystemAccounts = [
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply packages/server/tests/acceptance/rpc_batch1.spec.ts]: Due to [this definition](https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hedera-node/docs/system-accounts-operations.md#definition), any account within the 0-1000 range is referred to as a system account. So I think `hederaSystemAccounts` technically works, but yeah I agree it's a bit misleading so I renamed it to `hederaReservedAccounts`. Let me know wyt

**Review by acuarica — APPROVED** [review]

File: packages/server/tests/acceptance/rpc_batch1.spec.ts [hunk 3]
```diff
@@ -992,6 +992,180 @@ describe('@api-batch-1 RPC Server Acceptance Tests', function () {
...
+          description: '0.0.59 system delete admin',
+          expectedError: 'INVALID_CONTRACT_ID',
+        },
+        {
+          address: '0x000000000000000000000000000000000000003c',
+          description: '0.0.60 system undelete admin',
+          expectedError: 'INVALID_CONTRACT_ID',
+        },
+
+        // system contracts (precompiles) (≤ 0.0.750) - should return INVALID_CONTRACT_ID
```
Comment (acuarica) [inline packages/server/tests/acceptance/rpc_batch1.spec.ts:1057]: nit: use "precompiles" only for proper Ethereum precompiles to avoid confusion

```suggestion
        // system contracts (≤ 0.0.750) - should return INVALID_CONTRACT_ID
```

File: packages/server/tests/acceptance/rpc_batch1.spec.ts [hunk 4]
```diff
@@ -992,6 +992,180 @@ describe('@api-batch-1 RPC Server Acceptance Tests', function () {
...
+      // https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hedera-node/docs/system-accounts-operations.md
+      const hederaReservedAccounts = [
+        // system accounts (≤ 0.0.750) - should return INVALID_CONTRACT_ID
+        {
+          address: '0x0000000000000000000000000000000000000002',
+          description: '0.0.2 treasury',
+          expectedError: 'INVALID_CONTRACT_ID',
+        },
+        {
+          address: '0x0000000000000000000000000000000000000003',
```
Comment (acuarica) [inline packages/server/tests/acceptance/rpc_batch1.spec.ts:1022]: should we include a section with Ethereum precompiles?, that is, addresses in the range `0x1` to `0xa` (and also `0xb` in the future)

**Review by acuarica — APPROVED** [review]
out of curiosity, what happens when we `sendRawTransaction` to one of the Ethereum precompiles but **without** sending HBARs? is it a valid scenario?

**Review by acuarica — APPROVED** [review]

**Review by simzzz — DISMISSED** [review]
LGTM

**Review by natanasow — APPROVED** [review]
